Publish documentation to UDR instead of the Consul repo

rmainwork opened the pull request:
### Changes proposed in this PR ###  
- Update `consul-k8s` to publish documentation to UDR instead of the Consul repo, following [the Consul migration to UDR](https://github.com/hashicorp/web-unified-docs/pull/1228)

### How I've tested this PR ###
I've run `go test` in the `hack/helm-reference` directory to ensure that the tests still pass

### How I expect reviewers to test this PR ###
run `make gen-helm-docs` and follow the [instructions in the README](https://github.com/hashicorp/consul-k8s/commit/d4cdae6772add714097fe32ed266ebdc98335f18) to build helm docs.

### Checklist ###
- [ ] Tests added
- [ ] [CHANGELOG entry added](https://github.com/hashicorp/consul-k8s/blob/main/CONTRIBUTING.md#adding-a-changelog-entry) 

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1211305784283870

rmainwork committed:
Rename docs path to something more meaningful

rmainwork committed:
Write docs to UDR instead of consul repo

rmainwork committed:
Update instructions for UDR instead of consul repo

rmainwork committed:
Merge branch 'main' into rm/fetch-docs-from-udr

rmainwork committed:
Fix incorrect error reporting code

Co-authored-by: Ruben Nic <RubenSandwich@users.noreply.github.com>

rmainwork commented on hack/helm-reference-gen/main.go:
```diff
@@ -118,8 +119,45 @@ func main() {
+	versionMetadataPath := filepath.Join(docsRepoPath, "app/api/versionMetadata.json")
+	versionMetadataBytes, err := os.ReadFile(versionMetadataPath)
+	if err != nil {
+		fmt.Println(err)
```
Good catch(pun not intended)!

boruszak reviewed (dismissed):
I checked out your branch and ran the instructions as given. The file generated and I was able to open a PR: https://github.com/hashicorp/web-unified-docs/pull/1307/files

The PR adds outdated Helm fields though - I'm guessing your consul-k8s fork is behind the main repo. I suggest merging this PR, and then I can sort out these issues on my end.

Thanks so much!

rmainwork committed:
Fail Early

Fail early if any of the following are true:
- versionMetadata is empty
- versionMetadata fails to parse
- latestVersion is not set

rmainwork commented:
@RubenSandwich I restructured the code a little to fail early in [7421aef](https://github.com/hashicorp/consul-k8s/pull/4953/commits/7421aef3dc993b8f817cc933e6ab804371c6ea1a). Let me know if that's more in line with what you had in mind?

@boruszak Just noticed that adding a new commit dismissed your review, sorry if it needs to be re-approved or messed something up

RubenSandwich reviewed (approved):
Looks good now. 👍🏻

boruszak reviewed (approved):
LGTM!

boruszak committed:
Merge branch 'main' into rm/fetch-docs-from-udr
